model/shared/src/main/scala/explore/model/GroupElement.scala

```diff
@@ -54,23 +58,50 @@ object GroupElement:
   private def groupObsOr[B](f: HCursor => Decoder.Result[B]) = (c: HCursor) =>
     f(c).map(_.asRight).orElse(c.get[GroupObs]("observation").map(_.asLeft))
 
+    /**
+     * Find all parents of this element in the given group list
+     */
+  def findParentGroupIds(
+    groups:    GroupList,
+    elementId: Either[Observation.Id, Group.Id]
+  ): List[Group.Id] = {
+    @tailrec
+    def go(
+      acc:     List[Group.Id],
+      current: Either[Observation.Id, Group.Id]
+    ): List[Group.Id] =
+      val parentGroup = groups.find(
```

Comment:
To find the parent node, are we scanning the whole list of groups on each iteration?

Reply:
Yes, if it's not in a tree structure, I think this is the fastest/only way. Since I imagine groups won't be _that_ deep, it won't be too many iterations. `IndexedKeyTree` is already used, but I can have a look at lifting it higher